server/api/utils/singletons/k8s.py

```diff
@@ -540,6 +540,49 @@ def delete_secrets(
         self.v1api.delete_namespaced_secret(secret_name, namespace)
         return mlrun.common.schemas.SecretEventActions.deleted
 
+    def create_confmap(self, name: str, data: dict, namespace="", labels=None):
+        namespace = self.resolve_namespace(namespace)
+
+        body = client.V1ConfigMap(
+            api_version="v1",
+            kind="ConfigMap",
+            metadata=client.V1ObjectMeta(name=name, labels=labels),
+            data=data,
+        )
+
+        try:
+            self.v1api.create_namespaced_config_map(namespace=namespace, body=body)
+        except ApiException as exc:
+            if exc.status == 409:
+                try:
+                    self.v1api.replace_namespaced_config_map(
+                        name, namespace=namespace, body=body
+                    )
+                except ApiException as exc2:
+                    logger.error(
+                        f"failed to replace k8s ConfigMap {name}: {mlrun.errors.err_to_str(exc2)}"
+                    )
```

Comment:
use structured log and not templated and use capital letter for first word
```suggestion
                    logger.error(
                        "Failed replacing k8s config map", name=name, exc=mlrun.errors.err_to_str(exc2)"
                    )
```